src/bio/bio_device.c

```diff
@@ -672,3 +674,204 @@ bio_dev_list(struct bio_xs_context *xs_ctxt, d_list_t *dev_list, int *dev_cnt)
 
 	return rc;
 }
+
+int
+bio_set_led_state(struct bio_xs_context *xs_ctxt, char *dev_traddr,
+		  uuid_t dev_uuid, const char *led_state)
+{
+	struct spdk_pci_addr	pci_addr;
+	struct spdk_pci_device *pci_device;
+	struct bio_bdev	       *bio_dev;
+	char			addr_buf[128];
+	enum spdk_vmd_led_state current_led_state;
+	int			new_led_state;
+	int			rc = 0;
+	bool			found = false;
+
+	/* Determine SPDK LED state based on led_state string */
+	if (strcasecmp(led_state, "identify") == 0)
+		new_led_state = SPDK_VMD_LED_STATE_IDENTIFY;
+	else if (strcasecmp(led_state, "on") == 0)
+		new_led_state = SPDK_VMD_LED_STATE_FAULT;
+	else if (strcasecmp(led_state, "fault") == 0)
+		new_led_state = SPDK_VMD_LED_STATE_FAULT;
+	else if (strcasecmp(led_state, "off") == 0)
+		new_led_state = SPDK_VMD_LED_STATE_OFF;
+	else {
+		D_ERROR("LED state is not valid or supported\n");
+		return -DER_NOSYS;
+	}
+
+	bio_dev = lookup_dev_by_id(dev_uuid);
+	if (bio_dev == NULL) {
+		D_ERROR("Failed to find dev "DF_UUID"\n",
+			DP_UUID(dev_uuid));
+		return -DER_NONEXIST;
+	}
+
```

Comment:
Get the traddr by:

struct bio_dev_info b_info = { };
rc = fill_in_traddr(&b_info, bio_dev->bb_name);

Then free the b_info.bdi_traddr after use.